lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1724,7 +1724,7 @@ protected function get_layout_styles( $block_metadata, $types = array() ) {
 										$spacing_rule['selector']
 									);
 								} else {
-									$format          = static::ROOT_BLOCK_SELECTOR === $selector ? '.%2$s %3$s' : '%1$s-%2$s %3$s';
+									$format          = static::ROOT_BLOCK_SELECTOR === $selector ? ':root :where(.%2$s %3$s)' : ':root :where(%1$s-%2$s %3$s)';
```

Comment:
The problem before was that this would end up being a selector like `.is-layout-flow > *` (specificity 0,1,0) but when prefixed with `.editor-styles-wrapper` in the non-iframed editor it becomes `.editor-styles-wrapper .is-layout-flow > *` (specificity 0,2,0). It was reported that this has caused some issues - https://github.com/WordPress/gutenberg/issues/53468#issuecomment-2241279024.

My idea to fix that is to make it output `:root :where(.is-layout-flow > *)` (still specificity 0,1,0), and when we prefix those kind of rules we generally just replace the `:root` part, so it becomes `.editor-styles-wrapper where(.is-layout-flow > *)` (keeping the specificity at 0,1,0).

The trouble with these kind of generative rules is that I'll need to check every selector it produces to make sure it's valid and retaining the same specificity.

An alternative to this is that we could possibly read the specificity of the selector on the fly (using something like https://www.npmjs.com/package/specificity) and if it's high enough already we could prefix with `:where( .editor-styles-wrapper )` instead of `.editor-styles-wrapper`.

Reply:
Maybe it's worth breaking this into a separate PR, since this shouldn't be dependent on the transformStyles changes.

Reply:
Splitting it out into a separate PR - #64076